src/game/AI/ScriptDevAI/scripts/northrend/ulduar/ulduar/boss_ignis.cpp

```diff
@@ -73,309 +78,295 @@ enum
-
-        DoScriptText(urand(0, 1) ? SAY_SLAY_1 : SAY_SLAY_2, m_creature);
+        if (summoned->GetEntry() == NPC_SCORCH)
+        {
+            summoned->CastSpell(nullptr, m_isRegularMode ? SPELL_SCORCH_AURA : SPELL_SCORCH_AURA_H, TRIGGERED_OLD_TRIGGERED);
+            m_creature->AddSummonForOnDeathDespawn(summoned->GetObjectGuid());
+        }
     }
 
-    void Aggro(Unit* /*pWho*/) override
+    void OnPassengerRide(Unit* passenger, bool boarded, uint8 seat) override
     {
-        if (m_pInstance)
-            m_pInstance->SetData(TYPE_IGNIS, IN_PROGRESS);
-
-        DoScriptText(SAY_AGGRO, m_creature);
+        if (!boarded || !passenger || seat != 2)
+            return;
+        DoCastSpellIfCan(passenger, SPELL_SLAG_POT, CAST_TRIGGERED);
     }
+};
 
-    void JustReachedHome() override
+/*######
+## npc_iron_construct
+######*/
+
+struct npc_iron_constructAI : public ScriptedAI
+{
+    npc_iron_constructAI(Creature* creature) : ScriptedAI(creature),
+    m_isRegularMode(creature->GetMap()->IsRegularDifficulty())
     {
-        if (m_pInstance)
-            m_pInstance->SetData(TYPE_IGNIS, FAIL);
+        AddCustomAction(0, true, [&]()
+        {
+            m_creature->SetImmuneToPlayer(false);
+            instance_ulduar* instance = dynamic_cast<instance_ulduar*>(m_creature->GetInstanceData());
+            if (!instance)
+                return;
+            Creature* ignis = instance->GetSingleCreatureFromStorage(NPC_IGNIS);
+            if (!ignis || !ignis->AI())
+                return;
+            ignis->CastSpell(m_creature, SPELL_STRENGTH, TRIGGERED_OLD_TRIGGERED);
+            if (ignis->GetVictim())
+                AttackStart(ignis->GetVictim());
+        }, TIMER_COMBAT_OOC);
     }
 
-    void JustSummoned(Creature* pSummoned) override
+    bool m_isRegularMode;
+    bool m_hasShattered;
+
+    void JustRespawned() override
     {
-        if (pSummoned->GetEntry() == NPC_SCORCH)
-            pSummoned->CastSpell(pSummoned, m_bIsRegularMode ? SPELL_SCORCH_AURA : SPELL_SCORCH_AURA_H, TRIGGERED_OLD_TRIGGERED);
+        m_hasShattered = false;
+        
+        DoCastSpellIfCan(nullptr, SPELL_STONED, CAST_TRIGGERED);
+        DoCastSpellIfCan(nullptr, SPELL_CONSTRUCT_HITTING_YA, CAST_TRIGGERED);
     }
 
-    // TODO: Use the vehicle boarding wrappers when they are implemented in core
-    void SpellHit(Unit* pCaster, const SpellEntry* pSpell) override
+    void EnterEvadeMode() override
     {
-        if (pCaster->GetTypeId() != TYPEID_PLAYER)
-            return;
-
-        // Handle the case when passenger is loaded to the second seat
-        if (pSpell->Id == SPELL_GRAB_POT)
-            DoCastSpellIfCan(pCaster, SPELL_SLAG_POT, CAST_TRIGGERED);
+        m_creature->SetRespawnDelay(30, true);
+        m_creature->ForcedDespawn();
     }
 
-    void UpdateAI(const uint32 uiDiff) override
+    void DamageTaken(Unit* dealer, uint32& damage, DamageEffectType /*damagetype*/, SpellEntry const* spellInfo) override
```

Comment:
Spell 62382 brittle has proc flags. You can do this identical code in the proc, where you can check if mob received 5000+ damage. OnProc aura script hook.